query/src/sessions/session_ref.rs

```diff
@@ -57,14 +56,15 @@ impl Drop for SessionRef {
 impl Session {
     pub fn destroy_session_ref(self: &Arc<Self>) {
         if self.ref_count.fetch_sub(1, Ordering::Release) == 1 {
-            std::sync::atomic::fence(Acquire);
+            std::sync::atomic::fence(Ordering::Acquire);
```

Comment:
I think the `fence` call can be removed from both here and the following method `increment_ref_count`.

An operation to the internally referenced `Arc` is safe because it is protected by the fence inside `Arc`.